File: java/ex4_1.java
abstract class List<T> {
    abstract T get(int idx);
}

class ArrayList<T> extends List<T> {
    T get(int idx) { return null; }
}

class BitVector extends List<Boolean> {
    Boolean get(int idx) { return true; }
}

class Example {
    public static void main(String[] args) {
        List<String> l1 = new ArrayList<String>();
        List<Boolean> l2 = new ArrayList<Boolean>();
        List<Boolean> l3 = new BitVector();
    }
}

File: cs/ex4_1.cs
abstract class List<T> {
    public abstract T get(int idx);
}

class ArrayList<T> : List<T> {
    T t;
    public override T get(int idx) { return this.t; }
}

class BitVector : List<bool> {
    public override bool get(int idx) { return true; }
}

class Example {
    public static void Main() {
        List<string> l1 = new ArrayList<string>();
        List<bool> l2 = new ArrayList<bool>();
        List<bool> l3 = new BitVector();
    }
}
